dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -1277,6 +1274,242 @@ public void flushVanityURLCacheRight() throws ServletException, IOException {
         checkFilter(host_2, vanityUrlUpdated, -1);
     }
 
+    /**
+     * Method to test: {@link ESContentletAPIImpl#checkin(Contentlet, User, boolean)}
+     * When:
+     * - Create a ContentType with publish date
+     * - Create two Language
+     * - Create a {@link Contentlet} with the first language and publish date for tomorrow
+     * - Create a {@link Contentlet} with the second language and publish date for after tomorrow
+     * Should: Each {@link Contentlet} be saved and index with the right publish date
+     *
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void differentVersionWithDifferentPublishDate() throws DotDataException, DotSecurityException {
+
+        final Language language1 = new LanguageDataGen().nextPersisted();
+        final Language language2 = new LanguageDataGen().nextPersisted();
```

Comment:
Shouldn't we delete all the test languages in the test methods after they run? It seems like we'll end up with several tests languages, which might affect other tests in the end.

Reply:
done https://github.com/dotCMS/core/pull/22096/commits/9f0e7869e66bb92bf31604b93a882ffc887acd55#diff-093de07a53a725b587963fd60c024f52e089c4f78e7ea88ba8e3a3a0cdaecab9R1330